src/kaleidoscope/device/dygma/Raise.cpp

```diff
@@ -0,0 +1,599 @@
+
+  if (col >= 8) {
+    rightHandMask.rows[row] |= 1 << (7 - (col - 8));
+  } else {
+    leftHandMask.rows[row] |= 1 << (7 - col);
+  }
+}
+
+void RaiseKeyScanner::unMaskKey(KeyAddr key_addr) {
+  if (!key_addr.isValid())
+    return;
+
+  auto row = key_addr.row();
+  auto col = key_addr.col();
+
+  if (col >= 8) {
+    rightHandMask.rows[row] &= ~(1 << (7 - (col - 8)));
+  } else {
+    leftHandMask.rows[row] &= ~(1 << (7 - col));
+  }
+}
+
+bool RaiseKeyScanner::isKeyMasked(KeyAddr key_addr) {
+  if (!key_addr.isValid())
+    return false;
+
+  auto row = key_addr.row();
+  auto col = key_addr.col();
+
+  if (col >= 8) {
+    return rightHandMask.rows[row] & (1 << (7 - (col - 8)));
+  } else {
+    return leftHandMask.rows[row] & (1 << (7 - col));
+  }
+}
+
+void RaiseKeyScanner::maskHeldKeys() {
+  memcpy(leftHandMask.rows, leftHandState.rows, sizeof(leftHandMask));
+  memcpy(rightHandMask.rows, rightHandState.rows, sizeof(rightHandMask));
+}
+
+bool RaiseKeyScanner::isKeyswitchPressed(KeyAddr key_addr) {
+  auto row = key_addr.row();
+  auto col = key_addr.col();
+
+  if (col >= 8) {
+    return (bitRead(rightHandState.rows[row], 15 - col) != 0);
+  } else {
+    return (bitRead(leftHandState.rows[row], col) != 0);
+  }
+}
+
+bool RaiseKeyScanner::wasKeyswitchPressed(KeyAddr key_addr) {
+  auto row = key_addr.row();
+  auto col = key_addr.col();
+
+  if (col >= 8) {
+    return (bitRead(previousRightHandState.rows[row], 15 - col) != 0);
+  } else {
+    return (bitRead(previousLeftHandState.rows[row], col) != 0);
+  }
+}
+
+uint8_t RaiseKeyScanner::pressedKeyswitchCount() {
+  return __builtin_popcountl(leftHandState.all) + __builtin_popcountl(rightHandState.all);
+}
+
+uint8_t RaiseKeyScanner::previousPressedKeyswitchCount() {
+  return __builtin_popcountl(previousLeftHandState.all) + __builtin_popcountl(previousRightHandState.all);
+}
+
+void RaiseKeyScanner::setKeyscanInterval(uint8_t interval) {
+  RaiseHands::leftHand.setKeyscanInterval(interval);
+  RaiseHands::rightHand.setKeyscanInterval(interval);
+}
+
+void RaiseKeyScanner::setup() {
+  // first set all pins to outputs and low - EMC considerations
+  // not serial pins
+  for (int pin = 2; pin < 27; pin ++) {
+    pinMode(pin, OUTPUT);
+    digitalWrite(pin, LOW);
+  }
+  // not USB pins
+  for (int pin = 30; pin < 43; pin ++) {
+    pinMode(pin, OUTPUT);
+    digitalWrite(pin, LOW);
+  }
+}
+
+void RaiseKeyScanner::reset() {
+  leftHandState.all = 0;
+  rightHandState.all = 0;
+  kaleidoscope::hid::releaseAllKeys();
+  kaleidoscope::hid::sendKeyboardReport();
+}
+
+/********* Hardware plugin *********/
+void Raise::setup() {
+  RaiseHands::setup();
+  KeyScanner::setup();
+  LEDDriver::setup();
+
+  // initialise Wire of scanner - have to do this here to avoid problem with
+  // static object intialisation ordering
+  twi_init(I2C_CLOCK_KHZ);
+
+  RaiseHands::initializeSides();
+}
+
+void Raise::side::prepareForFlash() {
+  Wire.end();
+
+  power(LOW);
```

Comment:
"power" as a function name feels like it could be more evocative.

Reply:
Should I go with `disablePower()` and `enablePower()` instead?